### PR TITLE
Us20 implement manageBadgesEndpointTest

### PR DESCRIPTION
**Pull request by AmineChalouati**

### CODE REVIEW

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * Testing the Status Code
+     **/
+    @Test
+    public void testgetAllBadges() throws Exception {
+
+        // Given
+        org.junit.Assert.assertTrue(new ArrayList().isEmpty());
+        LOG.info("****Initiation of the assert of badges List**** ", BadgeEntity.class);
+
+        HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/all");
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Ends with org.apache.http.conn.HttpHostConnectException connection refused. Are you sure that something is runing under localhost:8080 when tests are executed? Should foccus how to mock the request

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        assertNotEquals(request, new ArrayList().isEmpty());
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: What is the relation between an empty ArrayList and request?

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+        // When
+        HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
+        assertThat(
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Not reashed. Exceptions are rised before

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                httpResponse.getStatusLine().getStatusCode(),
+                equalTo(HttpStatus.SC_OK));
+
+        LOG.info("List of badges is OK ", BadgeEntity.class, httpResponse.getStatusLine());
+
+        assertNotEquals(HttpStatus.SC_NOT_FOUND, HttpStatus.SC_OK);
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Please remove this evident assert. The two constants are differents within and without the test

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        // Givenge
+        try {
+            String Badgename = RandomStringUtils.randomAlphabetic(8);
+            String jsonMimeType = "application/json";
+
+            // Given
+            HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/add" + Badgename);
+
+            // When
+            HttpResponse response = HttpClientBuilder.create().build().execute(request);
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Will rise an exception and the following parts will not be executed

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+
+            // Then
+            String mimeType = ContentType.getOrDefault(response.getEntity()).getMimeType();
+            assertEquals(jsonMimeType, jsonMimeType);
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Always true

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        String badgeTitle = RandomStringUtils.randomAlphabetic(8);
+        String jsonMimeType = "application/json";
+
+
+        try {
+            // Given
+            HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/delete" + badgeTitle);
+
+            // When
+            HttpResponse response = HttpClientBuilder.create().build().execute(request);
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Will rise an exception and the following parts will not be executed

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+
+            // Then
+            String mimeType = ContentType.getOrDefault(response.getEntity()).getMimeType();
+            assertEquals(jsonMimeType, jsonMimeType);
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Always true

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+     **/
+    @Test
+    public void testupdatebadge() throws Exception {
+
+        // Givenge
+        String badgeTitle = RandomStringUtils.randomAlphabetic(8);
+        String jsonMimeType = "application/json";
+        try {
+            // Given
+            HttpUriRequest request = new HttpGet("http://localhost:8080/rest/gamification/badges/update");
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: will rise exception

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+            // When
+            HttpResponse response = HttpClientBuilder.create().build().execute(request);
+
+
+            // Then
+            String mimeType = ContentType.getOrDefault(response.getEntity()).getMimeType();
+            assertEquals(jsonMimeType, jsonMimeType);
+            LOG.info("The update of the selected badge is completely done with success ", BadgeEntity.class, response.getStatusLine());
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: Always true

File: services/src/test/java/org/exoplatform/addons/gamification/test/AbstractServiceTest.java [hunk 11]
```diff
@@ -14,29 +15,31 @@
 import org.exoplatform.social.core.space.spi.SpaceService;
```
Comment (bourasom) [inline services/src/test/java/org/exoplatform/addons/gamification/test/AbstractServiceTest.java]: What is the purpose of this class?

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java (hunk 1 above)
Comment (AmineChalouati) [reply services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: PR updated

File: services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java (hunk 2 above)
Comment (AmineChalouati) [reply services/src/test/java/org/exoplatform/addons/gamification/test/rest/TestManageBadgesEndpoint.java]: PR updated

File: services/src/main/java/org/exoplatform/addons/gamification/rest/ManageBadgesEndpoint.java [hunk 12]
```diff
@@ -15,283 +27,253 @@
...
+                                    inputStream);
 
-                }
+            fileItem = fileService.writeFile(fileItem);
 
-                /** END upload */
-                badgeDTO.setIconFileId(fileItem.getFileInfo().getId());
+          }
+          /** END upload */
+          badgeDTO.setIconFileId(fileItem.getFileInfo().getId());
```
Comment (AmineChalouati) [inline services/src/main/java/org/exoplatform/addons/gamification/rest/ManageBadgesEndpoint.java]: @bourasom this is the changement in the webservice

**Review by bourasom — COMMENTED** [review]
LGTM

**Pull request merged**